util-core/src/main/scala/com/twitter/conversions/StorageUnitOps.scala

```diff
@@ -16,7 +18,12 @@ import com.twitter.util.StorageUnit
  */
 object StorageUnitOps {
 
-  implicit class RichStorageUnit(val numBytes: Long) extends AnyVal {
+  /**
+   * Forwarder for Int, as Scala 3.0 seems to not like the implicit conversion to Long.
+   */
+  implicit def numBytesFromInt(numBytes: Int): RichStorageUnit = new RichStorageUnit(numBytes.toLong)
```

Comment:
Do you know what the Scala 3 compiler doesn't like the implicit conversion from int to long? Perhaps a bug?

Reply:
I don't see it as a bug. In Scala 2 you can already force a warning/error with `-Wnumeric-widen` in most cases where 
`Long` would implicitly be treated as `Int`. I'm not sure if that includes extension methods but if not that's an oversight imo. 

I think not doing these unsafe conversions implicitly is a good thing.

Reply:
Tbh, I'm not even sure why it worked before. This doesn't work for example: https://scastie.scala-lang.org/W972j1shQkWo4hVrtcQCHQ

Reply:
Another option would be to write the extension methods in terms of `Numeric` but that has other problems like how to handle floating-point numbers correctly and losing the value class:

https://scastie.scala-lang.org/S9UEDrf4R02nklksX6Qfaw

I'd probably not do this for the Scala 3 migration at least, as the implicit conversion to `Long` is much simpler.